### DISCUSSION TITLE
How to develop in my local?

### DISCUSSION THREAD

**Discussion by deiby-dayans** [osiset/laravel-shopify#954] [Q&A]
Hey guys!

I already have some time looking around into the project. First I developed some feature in my local, but some of them prompted some issues regarding getDomain null error, so I decided to create an Heroku instance and the project is running well, but the development is taking time because of pushing to heroku takes time.

Could you please share a setup where I can develop on my local? I was trying to use ngrok but I got some "getDomain on null" error while using a specific route that contains a form.

Thanks,
Deiby

**Comment by jonathonbyrdziak** [comment]
I built this docker image to run laravel on my local env for development. Make sure you have docker and docker-compose installed locally.

https://github.com/merchantprotocol/docker-nginx-php7.4-fpm/blob/master/example/simple-docker-compose/docker-compose.yml

Drop this docker-compose.yml file into your apps root directory and then run `docker-compose up` as usual.

The container is running php7.4 with all the necessary extensions along with nginx. I'm using it on multiple php projects. This container includes a working port 443 for secure connections on your localhost.

The default nginx install will expect your webroot to have your index.php file. For laravel we want the public folder to be our webroot though. So to customize the nginx configs, create a new folder inside your project base directory called `nginx.d` and then copy these files https://github.com/merchantprotocol/docker-nginx-php7.4-fpm/tree/master/nginx/sites-enabled into that directory. Rename these files to `nginx.conf` and `nginx-ssl.conf` respectively.

run `docker-compose up`

then access https://localhost in your browser and you're good to go.

This docker container will mount your projects files into /var/ww/html and make them available via a browser.

You can use VSCODE IDE to mount to the container and develop inside of it if you dont want to install php on your local machine.

**Comment by moonpetaldev** [comment]
There is an entire [wiki page](https://github.com/osiset/laravel-shopify/wiki/Developing-Locally) on developing locally.
